Shared/Samples/Validate utility network topology/ValidateUtilityNetworkTopologyView.swift

```diff
@@ -0,0 +1,198 @@
+                        operationIsRunning = true
+                        defer { operationIsRunning = false }
+                        
+                        do {
+                            switch selectedOperation {
+                            case .setup:
+                                try await model.setup()
+                                
+                            case .getState:
+                                try await model.getState()
+                                
+                            case .trace:
+                                try await model.trace()
+                                
+                            case .validateNetworkTopology:
+                                guard let extent = visibleArea?.extent else { return }
+                                try await model.validate(forExtent: extent)
+                                
+                            case .selectFeature(let screenPoint):
+                                // Identify the tapped layers using the map view proxy.
+                                let identifyResults = try await mapViewProxy.identifyLayers(
+                                    screenPoint: screenPoint!,
+                                    tolerance: 5
+                                )
+                                model.selectFeature(from: identifyResults)
+                                
+                                if model.feature != nil {
+                                    // Present the sheet to edit the feature if one was selected.
+                                    editSheetIsPresented = true
+                                } else {
+                                    model.statusMessage = "No feature identified. Tap on a feature."
+                                }
+                                
+                            case .applyEdits:
+                                try await model.applyEdits()
+                                
+                            case .clearSelection:
+                                model.clearSelection()
+                                model.statusMessage = "Selection cleared."
+                            }
+                        } catch {
+                            model.statusMessage = selectedOperation.errorMessage
+                            self.error = error
+                        }
+                    }
+                    .task(id: editSheetIsPresented) {
+                        guard editSheetIsPresented,
+                              let featureCenter = model.feature?.geometry?.extent.center else { return }
+                        
+                        // Create an envelope from the screen's frame.
+                        let viewRect = geometryProxy.frame(in: .local)
+                        guard let viewExtent = mapViewProxy.envelope(
+                            fromViewRect: viewRect
+                        ) else { return }
```

Comment:
can you just track the MapView's visible area and keep that in a state variable? That would avoid the geometry reader.